nexus_constructor/instrument_view/instrument_view.py

```diff
@@ -211,24 +215,30 @@ def add_component(
         if geometry is None:
             return
 
-        mesh = OffMesh(geometry.off_geometry, self.component_root_entity, positions)
-        material = create_material(
-            QColor("black") if name != SAMPLE_NAME else QColor("red"),
-            QColor("grey"),
-            self.component_root_entity,
-            alpha=0.5 if name == SAMPLE_NAME else None,
-        )
-
-        self.component_entities[name] = create_qentity(
-            [mesh, material], self.component_root_entity
-        )
+        if nx_class == SOURCE_CLASS_NAME:
+            entity = NeutronSource(self.component_root_entity)
+            self.component_entities[name] = entity
+        else:
+            mesh = OffMesh(geometry.off_geometry, self.component_root_entity, positions)
+            material = create_material(
+                MATERIAL_COLORS.get(nx_class, QColor("black")),
+                QColor("grey"),
+                self.component_root_entity,
+                MATERIAL_ALPHA.get(nx_class, None),
+            )
+            self.component_entities[name] = create_qentity(
+                [mesh, material], self.component_root_entity
+            )
 
     def get_entity(self, component_name: str) -> Qt3DCore.QEntity:
         """
         Obtain the entity from the InstrumentView based on its name.
         """
         try:
-            return self.component_entities[component_name]
+            component = self.component_entities[component_name]
```

Comment:
Hmm.. I would prefer if the component_entities actually only contain just that. Do you really need the reference here to be of type NeutronSource? Otherwise these isinstance checks will grow in the repo.